src/AutoDiscoveredValuesTrait.php

```diff
@@ -25,7 +25,14 @@ public static function values(): array
         $enumType = static::class;
 
         if (!isset(self::$guessedValues[$enumType])) {
-            $values = (new \ReflectionClass($enumType))->getConstants();
+            $r = new \ReflectionClass($enumType);
+            $values = $r->getConstants();
+
+            if (PHP_VERSION_ID >= 70100) {
+                $values = array_filter($values, function ($k) use ($r) {
```

Comment:
Thanks. That made me think I should filter non-integer values too for flagged enums.